src/construction.cpp

```diff
@@ -1241,6 +1243,38 @@ bool construct::check_support( const tripoint_bub_ms &p )
     return num_supports >= 2;
 }
 
+bool construct::check_support_below( const tripoint_bub_ms &p )
+{
+    bool blocking_creature = g->get_creature_if( [&]( const Creature & creature ) {
+        return creature.pos() == p.raw();
+    } ) != nullptr;
+
+    map &here = get_map();
+    // These checks are based on check_empty_lite, but accept empty air and its associated
+    // ledge trap.
+    // - Check if there's nothing in the way. The "passable" check rejects tiles you can't
+    //   pass through, but we want air and water to be OK as well (that's what we want to bridge).
+    // - Apart from that, vehicles, items, creatures, and furniture also have to be absent.
+    // - Then we have traps, and, unfortunately, there are "ledge" traps on all the open
+    //   space tiles adjacent to passable tiles, so we can't just reject all traps outright,
+    //   but have to accept those.
+    if( !( here.passable( p.raw() ) || here.has_flag( ter_furn_flag::TFLAG_LIQUID, p ) ||
+           here.ter( p ) == t_open_air ) ||
```

Comment:
We also have `t_open_air_rooved_outside` and `t_open_air_rooved`. Mods also might add new types of open air terrain. Maybe replace `t_open_air` with `NO_FLOOR` flag?

Reply:
Sounds like a very good idea. I wasn't thrilled about looking for a specific terrain, and was of a mixed mind about not covering things like pits (reasonable to be able to, but not reasonable to have them auto filled when the terrain is smashed).
I'll test and return.

Reply:
I've used NO_FLOOR, but have ended up extending the pit detection with specific pit traps, some of which don't have hard coded names.

Tested by map generating pits (shallow and deep), and "upgrading" two deep ones to glass and spike traps respectively. Could place gangway over all of them. Also covered the mining bug hole in the basement as a test.